vignettes/ees-api-workflow.Rmd

```diff
@@ -0,0 +1,282 @@
+---
+title: "EES API workflow"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{EES API workflow}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+```{r setup}
+```
+
+The following article presents an example workflow of how, as an analyst, someone might
+connect to the explore education statistics (EES) API to collect data with the help of
+`eesyapi` in R.
+
+As an overview, the stages you will likely need to follow are:
+
+- Find the data set ID
+- Query the data set meta data to identify filter and indicator IDs
+- Design a query using the relevant IDs
+- Post the query to the API
+- Perform some post processing of the data
+
+## Finding a data set ID
+
+Only some data sets on EES are available via the EES API. This is to maintain 
+minimum standards around the API data catalogue.
+
+Assuming you know the data set within a publication that you would like to query, 
+you will first need to identify the data set ID code. There are two methods to do 
+this, either a) via the data set's page on the EES website or b) by querying the 
+API itself.
+
+### Identifying a data set ID via the API
+
+Finding a data set ID via the API is a two step process. First, you'll need to find the parent publication ID and second, you'll need to search the data catalogue within that publication for the data set.
+
+Querying the publication list:
+
+```
+eesyapi::get_publications()
+```
+
+This returns a list of all publications by id and title (along with slug, summary and date last published). 
+From this list, you can find the title of the publication that you're interested in and pick out the 
+associated publication ID.
+
+For example from the test list, we can pick out **"Attendance API data"**, which has an 
+id of **"b6d9ed96-be68-4791-abc3-08dcaba68c04"**. We can now use this in the function `eesyapi::get_data_catalogue()` to find all the data sets within that publication:
+
+```
+eesyapi::get_data_catalogue("b6d9ed96-be68-4791-abc3-08dcaba68c04")
+```
+
+This returns a data frame containing data set ids and titles (along with a summary, the status and information on what's in the latest version). The data set ids can be used as shown in the following sections to get a given data set's summary and meta data, and to query the data set itself.
+
+### Finding a data set ID on the EES website
+
+As with any other data set, you can find an API data set in the EES data catalogue. If you
+know the name, you can filter the catalogue on this as normal. Not all data sets are accessible via the API, so to filter down to just those that are, click the **"API data sets only"** option as shown in the image below.
+
+![Finding an API data set in the EES data catalogue](ees-data-catalogue-find-api-data-set.png)
```

Comment:
This wants a bit more space between the caption and the next paragraph 

![image](https://github.com/user-attachments/assets/a20ce584-e954-4348-bd5c-9bd2be949753)

Reply:
I'd like to do that with some custom css, but aware we want this draft ready for 15 mins time, so thinking of leaving this as part of follow on clean-up.